core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -116,22 +117,23 @@ const setRole = async function(browser, page, role) {
       'http://127.0.0.1:8181/admin#/roles', { waitUntil: networkIdle });
     await page.waitForSelector(updateFormName);
     await page.type(updateFormName, 'username1');
-    await page.select(roleSelect, role);
+    await page.select(roleSelect, 'string:' + role);
     await page.waitForSelector(updateFormSubmit);
     await page.click(updateFormSubmit);
-    await page.waitForSelector(statusMessage);
+    await page.waitForSelector(statusMessage, {visible: true});
     await page.waitForFunction(
-      'document.querySelector(' +
```

Comment:
Why was this change neccesary here? Is it needed to help fix the errors here? If so, what was the issue.

Reply:
querySelector() only returns the first item on a page. I figured that maybe it was catching only the previous status message and that's why puppeteer was waiting for this call for a long time. I changed it to querySelectorAll() to make sure we look at every status message that gets displayed.

Reply:
It's still not working though -- see https://github.com/oppia/oppia/pull/13066/checks?check_run_id=2826877856. Per discussion with @krishnarao22 will try a timeout since this is blocking lots of PRs.

Reply:
I found another way to do this by changing the polling to explicitly poll every 500 ms. Previously I think it was polling on animation changes which probably just weren't happening for us.

@SAEb-ai I'm not sure whether this might help with the flake you are facing, too?

Reply:
Just to make sure await `page.waitForFunction('document.querySelector(' `
is flaking on develop?

Reply:
The error that's happening pretty often is the one mentioned in the first comment of this PR (see e.g. https://github.com/oppia/oppia/runs/2828382023?check_suite_focus=true and https://github.com/oppia/oppia/runs/2804142302?check_suite_focus=true). On further investigation, this error shadowed the check here which does exhibit the mentioned timeout on PRs unrelated to lighthouse or the codebase in general (e.g. adding translations to the assets files).